system/lib/wasmfs/wasmfs.cpp

```diff
@@ -303,4 +294,38 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
 
   return FileTable::get().add(openFile);
 }
+
+__wasi_errno_t __wasi_fd_seek(__wasi_fd_t fd,
+                              __wasi_filedelta_t offset,
+                              __wasi_whence_t whence,
+                              __wasi_filesize_t* newoffset) {
+  auto openFile = FileTable::get()[fd];
+  if (!openFile) {
+    return __WASI_ERRNO_BADF;
+  }
+  auto lockedOpenFile = openFile.locked();
+
+  off_t position = offset;
+  if (whence == SEEK_CUR) {
+    position = lockedOpenFile.position() + offset;
+  } else if (whence == SEEK_END) {
+    // Only the open file stat is altered in seek. Locking the underlying data
+    // file here once is sufficient.
+    position = lockedOpenFile.getFile()->locked().size() + offset;
+  } else if (whence != SEEK_SET) {
+    return __WASI_ERRNO_INVAL;
+  }
```

Comment:
It would be good to handle invalid `whence` values as well.

Reply:
Yeah that makes sense. I was just trying save a few lines by having `position = offset` by default (i.e. SEEK_SET)